src/modules/gateway/selectors.ts

```diff
@@ -1,24 +1,38 @@
 import { isLoadingType } from '../loading/selectors'
-import { PurchaseStatus } from '../mana/types'
 import {
   MANA_FIAT_GATEWAY_PURCHASE_COMPLETED,
   OPEN_MANA_FIAT_GATEWAY_REQUEST
 } from './actions'
-import { ManaFiatGatewayState } from './reducer'
+import { GatewayState } from './reducer'
+import { Purchase, PurchaseStatus } from './types'
 
-export const getState: (state: any) => ManaFiatGatewayState = state =>
-  state.manaFiatGateway
+const sortByTimestamp = (a: Purchase, b: Purchase) =>
+  a.timestamp > b.timestamp ? -1 : 1
+
+export const getState: (state: any) => GatewayState = state => state.gateway
 
 export const getData = (state: any) => getState(state).data
 export const getLoading = (state: any) => getState(state).loading
 export const getError = (state: any) => getState(state).error
 
 export const getPurchases = (state: any) => getData(state).purchases
-export const getPendingPurchase = (state: any) =>
+export const getPendingManaPurchase = (state: any) =>
   getPurchases(state).find(
-    purchase => purchase.status === PurchaseStatus.PENDING
+    purchase => purchase.status === PurchaseStatus.PENDING && !purchase.nft
   )
 
+export const getNFTPurchase = (
+  state: any,
+  contractAddress: string,
+  tokenId: string
+) =>
+  getPurchases(state)
+    .sort(sortByTimestamp)
```

Comment:
Sort will modify the original array. 

Store it first in a different array,

```
const arr = [...getPurchases(state)]
```

and then sort `arr`

Reply:
Nice catch 💯